src/scrapers/visa-cal.ts

```diff
@@ -194,60 +195,71 @@ async function fetchTransactionsForAccount(page: Page, startDate: Moment, accoun
       page.waitForNavigation({ waitUntil: 'domcontentloaded' }),
       clickButton(page, buttonSelector),
     ]);
-    debug('find the billing date');
-    const billingDateLabel = await pageEval(page, billingLabelSelector, '', ((element) => {
-      return (element as HTMLSpanElement).innerText;
+    debug('check if month has no transactions');
+    const pageHasNoTransactions = await pageEval(page, noDataSelector, false, ((element) => {
+      const siteValue = ((element as HTMLSpanElement).innerText || '').replace(/[^ א-ת]/g, '');
+      return siteValue === 'לא נמצאו נתונים';
     }));
```

Comment:
@esakal, Note that `$eval` (used inside `pageEval`) [throws](https://pptr.dev/#?product=Puppeteer&version=v12.0.1&show=api-frameevalselector-pagefunction-args-1) if no element matches the selector causing the scraper to fail when a card **has** transactions.

Adding a `catch` that resolves to false will fix it.

```suggestion
    })).catch(() => false);
```

Reply:
@daniel-hauser did it fail to you? because the infrastructure already take care on that here https://github.com/eshaham/israeli-bank-scrapers/blob/243da809308191203796f307d910a03678c7a9dc/src/helpers/elements-interactions.ts#L87

Let me know?
Eran

Reply:
Yes, it failed with `Error: failed to find element matching selector "[id$=FormAreaNoBorder_FormArea_msgboxErrorMessages]"`.

You can ignore my comment as I now see you fixed the infra error test (from `elements` to `element`) in this PR.